app/models/additional_document_validation_request.rb

```diff
@@ -1,12 +1,43 @@
 # frozen_string_literal: true
 
 class AdditionalDocumentValidationRequest < ApplicationRecord
+  class UploadFilesError < RuntimeError; end
+
   include ValidationRequest
 
   belongs_to :planning_application
   belongs_to :user
-  belongs_to :new_document, optional: true, class_name: "Document"
+
+  has_many :documents, dependent: :destroy
 
   validates :document_request_type, presence: { message: "Please fill in the document request type." }
   validates :document_request_reason, presence: { message: "Please fill in the reason for this document request." }
+
+  def upload_files!(files)
+    transaction do
+      files.each do |file|
+        planning_application.documents.create!(file: file, additional_document_validation_request: self)
```

Comment:
probably need to amend an association so that I don't need to do `additional_document_validation_request: self` .